imap_processing/tests/idex/test_idex_l2a.py

```diff
@@ -165,3 +168,77 @@ def test_calculate_snr_warning(caplog):
         "Unable to find baseline noise" in message
         for message in caplog.text.splitlines()
     )
+
+
+@pytest.mark.filterwarnings("ignore:invalid")
+@pytest.mark.filterwarnings("ignore:overflow")
+def test_analyze_peaks_warning(caplog):
+    """Tests that analyze_peaks() throws warning if the emg curve fit fails."""
+    # Create a 2d list of peak indices
+    peaks = [[2, 3, 4]]
+    time = xr.DataArray(np.arange(10))
+    # When there is a flat signal for TOF, we expect the fit to fail and a
+    # warning to be logged.
+    tof = np.ones_like(time)
+    with caplog.at_level("WARNING"):
+        fit_params, area_under_curve = analyze_peaks(tof, time, 0, peaks)
+    assert any(
+        "Failed to fit EMG curve" in message for message in caplog.text.splitlines()
+    )
+
+    # The fit_params and area_under_curve arrays should be zero
+    assert np.all(fit_params == 0)
+    assert np.all(area_under_curve == 0)
+
+
+def test_emg():
+    """Tests that emg() calculates an expected exponentially modified gaussian"""
+    mu = 4
+    sigma = 2.0
+    lam = 1.0
+    time = xr.DataArray(np.arange(100))
+    # The scipy.stats.exponnorm function's location and scale parameters match directly
```

Comment:
Should we be using this in place of the one you wrote then? Just inverting the input parameters as you've shown here?